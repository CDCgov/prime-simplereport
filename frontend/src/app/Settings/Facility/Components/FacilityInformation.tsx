import React from "react";

import { stateCodes } from "../../../../config/constants";
import Dropdown from "../../../commonComponents/Dropdown";
import TextInput from "../../../commonComponents/TextInput";
import { FacilityErrors } from "../facilitySchema";
import { ValidateField } from "../FacilityForm";
import { getSubStrAfterChar } from "../../../utils/text";

interface Props {
  facility: Facility;
  updateFacility: (facility: Facility) => void;
  errors: FacilityErrors;
  validateField: ValidateField;
  newOrg?: boolean;
}

const FacilityInformation: React.FC<Props> = ({
  facility,
  updateFacility,
  errors,
  validateField,
  newOrg = false,
}) => {
  const onChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    let fieldName = getSubStrAfterChar(e.target.name, "-");
    updateFacility({ ...facility, [fieldName]: e.target.value });
  };

  return (
    <div className="usa-form usa-form--large">
      {newOrg && (
        <>
          <p>
            Please enter one facility to get started. You can add more after
            account setup.
          </p>
          <p>
            If you plan to upload your results in bulk, you can include
            additional facilities in your spreadsheets without adding them in
            SimpleReport.
          </p>
        </>
      )}
      <h2 className="font-heading-lg" style={{ margin: 0 }}>
        Testing facility information
      </h2>
      <TextInput
        label="Testing facility name"
        name="name"
        value={facility.name}
        required
        onChange={onChange}
        onBlur={() => {
          validateField("name");
        }}
        validationStatus={errors.name ? "error" : undefined}
        errorMessage={errors.name}
      />
      <TextInput
        label="Phone number"
        name="facility-phone"
        value={facility.phone}
        required
        onChange={onChange}
        onBlur={() => {
          validateField("phone");
        }}
        validationStatus={errors.phone ? "error" : undefined}
        errorMessage={errors.phone}
      />
      <TextInput
        label="Email"
        name="email"
        data-testid="facility-email"
        value={facility.email || ""}
        onChange={onChange}
        onBlur={() => {
          validateField("email");
        }}
        validationStatus={errors.email ? "error" : undefined}
        errorMessage={errors.email}
      />
      <TextInput
        label="Street address 1"
        name="facility-street"
        value={facility.street}
        required
        onChange={onChange}
        onBlur={() => {
          validateField("street");
        }}
        validationStatus={errors.street ? "error" : undefined}
        errorMessage={errors.street}
      />
      <TextInput
        label="Street address 2"
        name="facility-streetTwo"
        value={facility.streetTwo || ""}
        onChange={onChange}
      />
      <TextInput
        label="City"
        name="facility-city"
        value={facility.city || ""}
        onChange={onChange}
      />
      <TextInput
        label="ZIP code"
        name="facility-zipCode"
        value={facility.zipCode}
        required
        onChange={onChange}
        onBlur={() => {
          validateField("zipCode");
        }}
        validationStatus={errors.zipCode ? "error" : undefined}
        errorMessage={errors.zipCode}
        className="usa-input--medium"
      />
      <Dropdown
        label="State"
        name="facility-state"
        selectedValue={facility.state}
        options={stateCodes.map((c) => ({ label: c, value: c }))}
        defaultSelect
        required
        onChange={onChange}
        onBlur={() => {
          validateField("state");
        }}
        validationStatus={errors.state ? "error" : undefined}
        errorMessage={errors.state}
        selectClassName="usa-input--medium"
        data-testid="facility-state-dropdown"
      />
      <TextInput
        label="CLIA number"
        hintText={
          <a
            href="https://www.cdc.gov/clia/LabSearch.html#"
            target="_blank"
            rel="noopener noreferrer"
          >
            Find my CLIA
          </a>
        }
        name="cliaNumber"
        value={facility.cliaNumber}
        required
        onChange={onChange}
        onBlur={() => {
          validateField("cliaNumber");
        }}
        validationStatus={errors.cliaNumber ? "error" : undefined}
        errorMessage={errors.cliaNumber}
      />
    </div>
  );
};

export default FacilityInformation;
