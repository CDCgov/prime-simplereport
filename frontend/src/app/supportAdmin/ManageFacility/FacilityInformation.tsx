import React, { useState } from "react";

import Button from "../../commonComponents/Button/Button";
import Modal from "../../commonComponents/Modal";
import { OrganizationTypeEnum } from "../../signUp/Organization/utils";

import { ManageFacilityState } from "./ManageFacility";

type FacilityDataItemProps = {
  title: string;
  description: string;
};

const FacilityDataItem: React.FC<FacilityDataItemProps> = ({
  title,
  description,
}) => (
  <div className="margin-y-3">
    <h4 className="margin-y-1">{title}</h4>
    <p className="margin-y-1">{description}</p>
  </div>
);

export interface FacilityInformationProps {
  onFacilityDelete: () => void;
  manageFacilityState: ManageFacilityState;
}

const FacilityInformation: React.FC<FacilityInformationProps> = ({
  manageFacilityState,
  onFacilityDelete,
}) => {
  const OrgTypeDictionary = OrganizationTypeEnum as any;

  /**
   * Confirmation modal
   */
  const [showModal, setShowModal] = useState(false);
  const closeModal = () => setShowModal(false);

  const handleConfirmDeletion = () => {
    onFacilityDelete();
    closeModal();
  };

  const confirmationModal = (
    <Modal
      onClose={closeModal}
      title="Delete facility"
      contentLabel="Confirm delete facility"
      showModal={showModal}
    >
      <Modal.Header
        styleClassNames={"font-heading-lg margin-top-0 margin-bottom-205"}
      >
        Delete {manageFacilityState.facility?.name}
      </Modal.Header>
      <div className="border-top border-base-lighter margin-x-neg-205"></div>
      <p className="margin-top-4">
        Deleting {manageFacilityState.facility?.name} will remove any
        in-progress tests from the test queue. Users only added to this facility
        will lose access to SimpleReport until an organization admin reassigns
        them. Patients only added to this facility also need to be reassigned.
      </p>
      <p className="margin-top-3">
        There are{" "}
        <strong>{manageFacilityState.facility?.usersCount} users</strong> and{" "}
        <strong>{manageFacilityState.facility?.patientsCount} patients</strong>{" "}
        added only to this facility.
      </p>
      <div className="border-top border-base-lighter margin-x-neg-205"></div>
      <Modal.Footer
        styleClassNames={"display-flex flex-justify-end margin-top-205"}
      >
        <Button
          className="margin-right-205"
          variant="unstyled"
          label="No, go back"
          onClick={closeModal}
        />
        <Button
          className="margin-right-0"
          label="Yes, delete facility"
          onClick={handleConfirmDeletion}
        />
      </Modal.Footer>
    </Modal>
  );

  /**
   * HTML
   */
  return (
    <div className="prime-container card-container" aria-live={"polite"}>
      {manageFacilityState.facility ? (
        <>
          <div className="usa-card__header">
            <h2 className="desktop:grid-col-fill tablet:grid-col-fill mobile:grid-col-12 font-heading-lg margin-top-0 margin-bottom-0">
              {manageFacilityState.facility.name}
            </h2>
          </div>
          <div className="usa-card__body">
            <h3 className="margin-y-3">Facility information</h3>
            <FacilityDataItem
              title="Facility name"
              description={manageFacilityState.facility.name}
            />
            <FacilityDataItem
              title="Location"
              description={`${
                manageFacilityState.facility.city !== "" &&
                manageFacilityState.facility.city?.concat(", ")
              }${manageFacilityState.facility.state} ${
                manageFacilityState.facility.zipcode
              }`}
            />
            <FacilityDataItem
              title="Organization"
              description={manageFacilityState.facility.org}
            />
            <FacilityDataItem
              title="Organization type"
              description={
                OrgTypeDictionary[manageFacilityState.facility.orgType]
              }
            />
            <h3 className="margin-top-6">Facility controls</h3>
            <div className="margin-y-3">
              <h4 className="margin-y-1">Delete facility</h4>
              <div className="grid-row width-full flex-justify">
                <p className=" desktop:grid-col-8 tablet:grid-col-8 mobile:grid-col-12  margin-y-1 text-italic text-base-darker line-height-sans-3">
                  Any in-progress tests from this facility will be removed, and
                  an organization admin will need to reassign users and patients
                  only added to this facility.
                </p>
                <div className="desktop:grid-col-auto tablet:grid-col-auto mobile:grid-col-12 flex-align-end">
                  <Button
                    variant={"outline"}
                    onClick={() => setShowModal(true)}
                    ariaLabel={`Delete facility ${manageFacilityState.facility.name}`}
                  >
                    Delete facility
                  </Button>
                </div>
              </div>
            </div>
          </div>
        </>
      ) : (
        <p className={"usa-card__body padding-y-3"}>No facility selected. </p>
      )}
      {confirmationModal}
    </div>
  );
};

export default FacilityInformation;
