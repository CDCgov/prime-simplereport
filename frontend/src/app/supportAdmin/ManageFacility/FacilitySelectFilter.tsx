import { faSlidersH } from "@fortawesome/free-solid-svg-icons";
import React, { Ref } from "react";
import { ComboBoxRef } from "@trussworks/react-uswds";

import ComboBox from "../../commonComponents/ComboBox";
import Button from "../../commonComponents/Button/Button";
import SupportHomeLink from "../SupportHomeLink";
import { Option } from "../../commonComponents/Dropdown";

import { ManageFacilityState } from "./ManageFacility";

export interface FacilitySelectFilterProps {
  organizationOptions: Option[];
  facilityOptions: Option[];
  manageFacilityState: ManageFacilityState;
  onClearFilter: () => void;
  onSelectOrg: (selectedOrg: string | undefined) => void;
  onSelectFacility: (selectedFacility: string | undefined) => void;
  onSearch: (e: any) => void;
  loading: boolean;
  facilityRef: Ref<ComboBoxRef> | undefined;
  orgRef: Ref<ComboBoxRef> | undefined;
}

const FacilitySelectFilter: React.FC<FacilitySelectFilterProps> = ({
  organizationOptions,
  facilityOptions,
  manageFacilityState,
  onClearFilter,
  onSelectOrg,
  onSelectFacility,
  onSearch,
  loading,
  facilityRef,
  orgRef,
}) => {
  /**
   * HTML
   */
  return (
    <div className="prime-container card-container padding-bottom-3">
      <div className="usa-card__header">
        <div className="width-full">
          <SupportHomeLink />
          <div className="grid-row width-full margin-top-1">
            <h1 className="desktop:grid-col-fill tablet:grid-col-fill mobile:grid-col-12 font-heading-lg margin-bottom-0">
              Manage facility
            </h1>
            <div className="desktop:grid-col-auto tablet:grid-col-auto mobile:grid-col-12 margin-top-2 tablet:margin-top-0">
              <Button
                icon={faSlidersH}
                disabled={manageFacilityState.orgId === undefined}
                onClick={onClearFilter}
                ariaLabel="Clear facility selection filters"
              >
                Clear filters
              </Button>
            </div>
          </div>
        </div>
      </div>
      <div
        role="search"
        className="bg-base-lightest padding-left-3 padding-right-3 padding-bottom-1"
      >
        <div className="grid-row grid-gap padding-bottom-2 flex-align-end">
          <div
            data-testid={"org-selection-container"}
            className="desktop:grid-col-4 tablet:grid-col-4 mobile:grid-col-1 margin-top-1em"
          >
            <ComboBox
              name={"Organization"}
              id={"manage-facility-org-select"}
              options={organizationOptions}
              onChange={(val) => {
                onSelectOrg(val);
              }}
              disabled={loading}
              ref={orgRef}
              required
            />
          </div>
          <div
            data-testid={"facility-selection-container"}
            className="desktop:grid-col-4 tablet:grid-col-4 mobile:grid-col-1 margin-top-1em"
          >
            <ComboBox
              name={"Facility"}
              id={"manage-facility-facility-select"}
              options={facilityOptions}
              onChange={(val) => onSelectFacility(val)}
              disabled={loading}
              ref={facilityRef}
              required
            />
          </div>
          <div className="desktop:grid-col-4 tablet:grid-col-4 mobile:grid-col-1 ">
            <Button
              onClick={onSearch}
              disabled={manageFacilityState.facilityId === undefined}
              ariaLabel="Search facility selection"
            >
              Search
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default FacilitySelectFilter;
