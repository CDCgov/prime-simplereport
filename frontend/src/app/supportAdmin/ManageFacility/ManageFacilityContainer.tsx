import React, { useEffect, useState } from "react";

import { useDocumentTitle } from "../../utils/hooks";
import PageContainer from "../PageContainer";
import Dropdown, { Option } from "../../commonComponents/Dropdown";
import {
  useGetAllOrganizationsQuery,
  useGetFacilitiesByOrgIdLazyQuery,
} from "../../../generated/graphql";

const ManageFacilityContainer = () => {
  useDocumentTitle("Manage Facility");

  /**
   * Initialize (org dropdown)
   */
  const [orgId, setOrgId] = useState<string>("");
  const { loading: loadingOrgs, data: orgResponse } =
    useGetAllOrganizationsQuery();
  const orgOptions: Option[] =
    orgResponse?.organizations?.map((org) => ({
      value: org.externalId,
      label: org.name,
    })) ?? [];

  /**
   * Fetch facilities by org
   */
  const [queryGetFacilitiesByOrgId, { data: facilitiesResponse }] =
    useGetFacilitiesByOrgIdLazyQuery();

  useEffect(() => {
    if (orgId !== "") {
      queryGetFacilitiesByOrgId({ variables: { orgId } });
    }
  }, [orgId, queryGetFacilitiesByOrgId]);

  const facilitiesOptions: Option[] =
    facilitiesResponse?.organization?.facilities.map((facility) => ({
      value: facility.id,
      label: facility.name,
    })) ?? [];

  /**
   * Retrieve facility
   */

  /**
   * Search controls
   */
  const searchControls = (
    <div className="grid-row grid-gap padding-bottom-2">
      <div className="desktop:grid-col-4 tablet:grid-col-4 mobile:grid-col-1">
        <Dropdown
          label="Organization"
          options={[
            {
              label: "- Select -",
              value: "",
            },
            ...orgOptions,
          ]}
          onChange={(e) => {
            setOrgId(e.target.value);
          }}
          selectedValue={orgId}
          disabled={loadingOrgs}
        />
      </div>
      <div className="desktop:grid-col-4 tablet:grid-col-4 mobile:grid-col-1">
        <Dropdown
          label="Facility"
          options={[
            {
              label: "- Select -",
              value: "",
            },
            ...facilitiesOptions,
          ]}
          onChange={() => {}}
          selectedValue={""}
          disabled={loadingOrgs || orgId === ""}
        />
      </div>
    </div>
  );

  /**
   * HTML
   */
  return (
    <PageContainer title="Manage facility" controls={searchControls}>
      <p className={"text-center padding-3"}>No facility selected</p>
    </PageContainer>
  );
};

export default ManageFacilityContainer;
