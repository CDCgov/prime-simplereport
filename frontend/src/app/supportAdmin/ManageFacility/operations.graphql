query GetAllOrganizations {
    organizations(identityVerified: true){
        id
        name
    }
}

query GetFacilitiesByOrgId($orgId: ID!) {
    organization(id: $orgId){
        id
        externalId
        name
        type
        facilities {
            name
            id
            city
            state
            zipCode
        }
    }
}


query GetFacilityStats($facilityId: ID!) {
    facilityStats(facilityId: $facilityId){
        usersSingleAccessCount
        patientsSingleAccessCount
    }
}

mutation DeleteFacility($facilityId: ID!) {
    markFacilityAsDeleted(
        facilityId: $facilityId
        deleted: true
    )
}