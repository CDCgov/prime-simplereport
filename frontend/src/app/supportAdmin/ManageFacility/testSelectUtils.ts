import { screen, within } from "@testing-library/react";

export const getOrgComboBoxElements = () => {
  const orgSelectionDiv = screen.getByTestId("org-selection-container");
  const orgComboBoxInput = screen.getByLabelText(/organization/i);
  const orgComboBoxList = within(orgSelectionDiv).getByTestId(
    "combo-box-option-list"
  );
  return [orgComboBoxInput, orgComboBoxList] as const;
};

export const getFacilityComboBoxElements = () => {
  const facilitySelectionDiv = screen.getByTestId(
    "facility-selection-container"
  );
  const facilityComboBoxInput = screen.getByLabelText(/testing facility/i);
  const facilityComboBoxList = within(facilitySelectionDiv).getByTestId(
    "combo-box-option-list"
  );
  return [facilityComboBoxInput, facilityComboBoxList] as const;
};
