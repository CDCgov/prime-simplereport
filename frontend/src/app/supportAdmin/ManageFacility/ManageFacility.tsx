import React, { useState } from "react";

import { Option } from "../../commonComponents/Dropdown";
import {
  useDeleteFacilityMutation,
  useGetAllOrganizationsQuery,
  useGetFacilitiesByOrgIdLazyQuery,
  useGetFacilityStatsLazyQuery,
} from "../../../generated/graphql";
import { useDocumentTitle } from "../../utils/hooks";
import { showSuccess } from "../../utils/srToast";

import FacilitySelectFilter from "./FacilitySelectFilter";
import FacilityInformation from "./FacilityInformation";

type Facility = {
  id: string;
  name: string;
  city: string | null | undefined;
  state: string;
  zipcode: string;
  org: string;
  orgType: string;
  usersCount: number;
  patientsCount: number;
};

export interface ManageFacilityState {
  orgId: string;
  facilityId: string;
  facility: Facility | undefined;
}

const initialState: ManageFacilityState = {
  orgId: "",
  facilityId: "",
  facility: undefined,
};

const ManageFacility = () => {
  useDocumentTitle("Manage Facility");
  const [localState, updateLocalState] =
    useState<ManageFacilityState>(initialState);

  /**
   * Fetch organizations (on initial load)
   */
  const { data: orgResponse, loading: loadingOrgs } =
    useGetAllOrganizationsQuery();

  const orgOptions: Option[] =
    orgResponse?.organizations?.map((org) => ({
      value: org.id,
      label: org.name,
    })) ?? [];

  /**
   * Fetch facilities
   */
  const [
    queryGetFacilitiesByOrgId,
    { data: facilitiesResponse, loading: loadingFacilities },
  ] = useGetFacilitiesByOrgIdLazyQuery();

  const facilitiesOptions: Option[] =
    facilitiesResponse?.organization?.facilities.map((facility) => ({
      value: facility.id,
      label: facility.name,
    })) ?? [];

  /**
   * Fetch facility stats
   */
  const [queryGetFacilityStats] = useGetFacilityStatsLazyQuery();

  /**
   * Facility select filter handlers
   */
  function handleSelectOrganization(e: React.ChangeEvent<HTMLSelectElement>) {
    const selectedOrg = e.target.value;
    if (selectedOrg !== "") {
      queryGetFacilitiesByOrgId({
        variables: { orgId: selectedOrg },
        fetchPolicy: "no-cache",
      }).then();
    }

    updateLocalState({ ...initialState, orgId: selectedOrg });
  }
  async function handleSelectFacility(e: React.ChangeEvent<HTMLSelectElement>) {
    const facilityId = e.target.value;

    if (facilityId === "") {
      updateLocalState((prevState) => ({
        ...prevState,
        facilityId: "",
        facility: undefined,
      }));
    } else {
      const selectedFacility =
        facilitiesResponse?.organization?.facilities?.filter(
          (f) => f.id === e.target.value
        )?.[0];

      const facilityStats = await queryGetFacilityStats({
        fetchPolicy: "no-cache",
        variables: { facilityId },
      }).then((response) => response.data?.facilityStats);

      updateLocalState((prevState) => ({
        orgId: prevState.orgId,
        facilityId: facilityId,
        facility: {
          city: selectedFacility?.city,
          state: selectedFacility?.state || "",
          zipcode: selectedFacility?.zipCode || "",
          id: selectedFacility?.id || "",
          name: selectedFacility?.name || "",
          org: facilitiesResponse?.organization?.name || "",
          orgType: facilitiesResponse?.organization?.type || "",
          usersCount: facilityStats?.usersSingleAccessCount || 0,
          patientsCount: facilityStats?.patientsSingleAccessCount || 0,
        },
      }));
    }
  }

  function handleClearFilter() {
    updateLocalState(initialState);
  }

  /**
   * Delete facility
   */
  const [deleteFacilityMutation] = useDeleteFacilityMutation();
  function handleDeleteFacility() {
    deleteFacilityMutation({
      variables: { facilityId: localState.facilityId },
    }).then(() => {
      showSuccess(
        "",
        `Facility ${localState.facility?.name} successfully deleted`
      );
      handleClearFilter();
    });
  }

  /**
   * HTML
   */
  return (
    <div className="prime-home flex-1">
      <div className="grid-container">
        <FacilitySelectFilter
          onSelectFacility={handleSelectFacility}
          onClearFilter={handleClearFilter}
          onSelectOrg={handleSelectOrganization}
          facilityOptions={facilitiesOptions}
          organizationOptions={orgOptions}
          manageFacilityState={localState}
          loading={loadingOrgs || loadingFacilities}
        />
        <FacilityInformation
          manageFacilityState={localState}
          onFacilityDelete={handleDeleteFacility}
        />
      </div>
    </div>
  );
};

export default ManageFacility;
