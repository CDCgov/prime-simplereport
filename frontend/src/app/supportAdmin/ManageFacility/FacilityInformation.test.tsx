import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import FacilityInformation from "./FacilityInformation";
import { initialState, ManageFacilityState } from "./ManageFacility";

describe("Facility Information", () => {
  const handleFacilityDelete = jest.fn();
  const mockManageFacilityState: ManageFacilityState = {
    orgId: "123",
    facilityId: "123",
    facility: {
      city: "New York",
      state: "NY",
      zipcode: "12345",
      id: "123",
      name: "The new center",
      org: "Wild trees camp",
      orgType: "school",
      usersCount: 0,
      patientsCount: 0,
    },
  };

  it("renders no selected facility", () => {
    const { container } = render(
      <FacilityInformation
        onFacilityDelete={handleFacilityDelete}
        manageFacilityState={{ ...initialState }}
      />
    );
    expect(screen.getByText(/No facility selected/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders facility information", () => {
    render(
      <FacilityInformation
        onFacilityDelete={handleFacilityDelete}
        manageFacilityState={{ ...mockManageFacilityState }}
      />
    );
    expect(document.body).toMatchSnapshot();
  });

  it("Confirmation modal and go back", async () => {
    render(
      <FacilityInformation
        onFacilityDelete={handleFacilityDelete}
        manageFacilityState={{ ...mockManageFacilityState }}
      />
    );
    const deleteFacilityBtn = screen.getByRole("button", {
      name: /delete facility the new center/i,
    });
    fireEvent.click(deleteFacilityBtn);
    await screen.findByRole("heading", { name: /delete the new center/i });
    expect(document.body).toMatchSnapshot();
    const noGoBackBtn = screen.getByRole("button", { name: /no, go back/i });
    fireEvent.click(noGoBackBtn);
    await waitFor(() =>
      expect(
        screen.queryByRole("heading", { name: /delete the new center/i })
      ).not.toBeInTheDocument()
    );
  });

  it("Confirmation modal and yes delete", async () => {
    render(
      <FacilityInformation
        onFacilityDelete={handleFacilityDelete}
        manageFacilityState={{ ...mockManageFacilityState }}
      />
    );
    const deleteFacilityBtn = screen.getByRole("button", {
      name: /delete facility the new center/i,
    });

    fireEvent.click(deleteFacilityBtn);
    await screen.findByRole("heading", { name: /delete the new center/i });

    const yesDeleteBtn = screen.getByRole("button", {
      name: /yes, delete facility/i,
    });
    fireEvent.click(yesDeleteBtn);
    await waitFor(() =>
      expect(
        screen.queryByRole("heading", { name: /delete the new center/i })
      ).not.toBeInTheDocument()
    );
    expect(handleFacilityDelete).toHaveBeenCalled();
  });
});
