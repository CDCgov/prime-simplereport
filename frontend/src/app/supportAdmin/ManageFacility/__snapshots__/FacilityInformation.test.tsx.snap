// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facility Information Confirmation modal and go back 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div>
    <div
      aria-live="polite"
      class="prime-container card-container"
    >
      <div
        class="usa-card__header"
      >
        <h2
          class="desktop:grid-col-fill tablet:grid-col-fill mobile:grid-col-12 font-heading-lg margin-top-0 margin-bottom-0"
        >
          The new center
        </h2>
      </div>
      <div
        class="usa-card__body"
      >
        <h3
          class="margin-y-3"
        >
          Facility information
        </h3>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Facility name
          </h4>
          <p
            class="margin-y-1"
          >
            The new center
          </p>
        </div>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Location
          </h4>
          <p
            class="margin-y-1"
          >
            New York, NY 12345
          </p>
        </div>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Organization
          </h4>
          <p
            class="margin-y-1"
          >
            Wild trees camp
          </p>
        </div>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Organization type
          </h4>
          <p
            class="margin-y-1"
          />
        </div>
        <h3
          class="margin-top-6"
        >
          Facility controls
        </h3>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Delete facility
          </h4>
          <div
            class="grid-row width-full flex-justify"
          >
            <p
              class=" desktop:grid-col-8 tablet:grid-col-8 mobile:grid-col-12  margin-y-1 text-italic text-base-darker line-height-sans-3"
            >
              Any in-progress tests from this facility will be removed, and an organization admin will need to reassign users and patients only added to this facility.
            </p>
            <div
              class="desktop:grid-col-auto tablet:grid-col-auto mobile:grid-col-12 flex-align-end"
            >
              <button
                aria-label="Delete facility The new center"
                class="usa-button usa-button--outline"
                type="button"
              >
                Delete facility
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal--basic"
  >
    <div
      class="ReactModal__Overlay prime-modal-overlay display-flex flex-align-center flex-justify-center"
    >
      <div
        aria-label="Confirm delete facility"
        aria-modal="true"
        class="ReactModal__Content"
        role="dialog"
        style="position: initial; top: 40px; left: 40px; right: 40px; bottom: 40px; border: 0px; background: rgb(255, 255, 255); overflow: auto; border-radius: 4px; outline: none; padding: 20px;"
        tabindex="-1"
      >
        <div
          class="modal__container"
        >
          <button
            class="modal__close-button"
            style="cursor: pointer; z-index: 1;"
          >
            <img
              alt="Close"
              class="modal__close-img"
              src="close.svg"
            />
          </button>
          <main>
            <div
              class="modal__content grid-row"
            >
              <div
                class="grid-col"
              >
                <h1
                  class="modal__heading font-heading-lg margin-top-0 margin-bottom-205"
                >
                  Delete 
                  The new center
                </h1>
                <div
                  class="border-top border-base-lighter margin-x-neg-205"
                />
                <p
                  class="margin-top-4"
                >
                  Deleting 
                  The new center
                   will remove any in-progress tests from the test queue. Users only added to this facility will lose access to SimpleReport until an organization admin reassigns them. Patients only added to this facility also need to be reassigned.
                </p>
                <p
                  class="margin-top-3"
                >
                  There are 
                  <strong>
                    0
                     users
                  </strong>
                   
                  and
                   
                  <strong>
                    0
                     patients
                  </strong>
                   
                  added only to this facility.
                </p>
                <div
                  class="border-top border-base-lighter margin-x-neg-205"
                />
                <div
                  class="modal__footer display-flex flex-justify-end margin-top-205"
                >
                  <button
                    class="usa-button usa-button--unstyled margin-right-205"
                    type="button"
                  >
                    No, go back
                  </button>
                  <button
                    class="usa-button margin-right-0"
                    type="button"
                  >
                    Yes, delete facility
                  </button>
                </div>
              </div>
            </div>
          </main>
        </div>
      </div>
    </div>
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Facility Information renders facility information 1`] = `
<body>
  <div>
    <div
      aria-live="polite"
      class="prime-container card-container"
    >
      <div
        class="usa-card__header"
      >
        <h2
          class="desktop:grid-col-fill tablet:grid-col-fill mobile:grid-col-12 font-heading-lg margin-top-0 margin-bottom-0"
        >
          The new center
        </h2>
      </div>
      <div
        class="usa-card__body"
      >
        <h3
          class="margin-y-3"
        >
          Facility information
        </h3>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Facility name
          </h4>
          <p
            class="margin-y-1"
          >
            The new center
          </p>
        </div>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Location
          </h4>
          <p
            class="margin-y-1"
          >
            New York, NY 12345
          </p>
        </div>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Organization
          </h4>
          <p
            class="margin-y-1"
          >
            Wild trees camp
          </p>
        </div>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Organization type
          </h4>
          <p
            class="margin-y-1"
          />
        </div>
        <h3
          class="margin-top-6"
        >
          Facility controls
        </h3>
        <div
          class="margin-y-3"
        >
          <h4
            class="margin-y-1"
          >
            Delete facility
          </h4>
          <div
            class="grid-row width-full flex-justify"
          >
            <p
              class=" desktop:grid-col-8 tablet:grid-col-8 mobile:grid-col-12  margin-y-1 text-italic text-base-darker line-height-sans-3"
            >
              Any in-progress tests from this facility will be removed, and an organization admin will need to reassign users and patients only added to this facility.
            </p>
            <div
              class="desktop:grid-col-auto tablet:grid-col-auto mobile:grid-col-12 flex-align-end"
            >
              <button
                aria-label="Delete facility The new center"
                class="usa-button usa-button--outline"
                type="button"
              >
                Delete facility
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal--basic"
  />
</body>
`;

exports[`Facility Information renders no selected facility 1`] = `
<div>
  <div
    aria-live="polite"
    class="prime-container card-container"
  >
    <p
      class="usa-card__body padding-y-3"
    >
      No facility selected. 
    </p>
  </div>
</div>
`;
